pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -72,6 +69,21 @@ func (gce *GCECloud) ensureExternalLoadBalancer(clusterName, clusterID string, a
 	glog.V(2).Infof("EnsureLoadBalancer(%v, %v, %v, %v, %v, %v, %v)",
 		loadBalancerName, gce.region, requestedIP, portStr, hostNames, serviceName, apiService.Annotations)
 
+	lbRefStr := fmt.Sprintf("%v(%v)", loadBalancerName, serviceName)
+	// Check the current and the desired network tiers. If they do not match,
+	// tear down the existing resources -- delete the forwarding rule, and
+	// release the IP.
+	netTier, err := gce.getServiceNetworkTier(apiService)
```

Comment:
Could we also gate network tier outside of `getServiceNetworkTier()`? Like `reconcileXXXNetworkTiers()` below, so that we don't trigger the logic at all.

Reply:
I did that intentionally since `getServiceNetworkTier()` should always return the default tier if the feature is disabled. If I could, I'd actually prefer gating it only at the gce API calls so that we can reuse the same flow when the feature graduates to Beta. Otherwise, the beta code will be completely new code which needs to be tested all over again. I think the current implementation provides sufficient "gating" but feel free to disagree :-)

Reply:
Humm...I think I some how mixed the concepts of gating alpha/beta GCE API and gating the actual feature. IMHO:
- Gate alpha/beta API means no calls to them.
- Gate the feature means no triggers to the logic.

Given that the GCE provider gate is about gating API but not feature, I think it makes sense to do it this way...

Reply:
> I think the current implementation provides sufficient "gating" but feel free to disagree :-)

But yeah technically this is sufficient :)